lora/paho/sub.go

```diff
@@ -0,0 +1,52 @@
+package paho
+
+// LoraSubscribe subscribe to lora server messages
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/lora"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	nats "github.com/nats-io/go-nats"
+)
+
+const loraServerTopic = "application/+/device/+/rx"
+
+type pubsub struct {
+	nc     *nats.Conn
+	svc    lora.Service
+	logger logger.Logger
+}
+
+// Subscribe subscribes to the Lora MQTT message broker
+func Subscribe(svc lora.Service, mc mqtt.Client, nc *nats.Conn, log logger.Logger) error {
```

Comment:
Make factory method where you initialize `pubsub`, and separate subscribe method where you just pass subject (topic). This should be done in order to separate initialization and subscription logic. Also, you don't need NATS connection in this struct.